File: day_3/part_1.py
inputFile = open('input.txt').read().split('\n')

gamma = [0]*12

for line in inputFile:
    if len(line) == 0:
        break
    for i in range(0,12):
        if int(line[i]):
            gamma[i] += 1
        else:
            gamma[i] -= 1

gamma = list(map(lambda i : "1" if i>0 else "0" , gamma))
epsilon = list(map(lambda i : "0" if i == "1" else "1", gamma))
gamma = int("".join(gamma), 2)
epsilon = int("".join(epsilon), 2)

print(gamma * epsilon)

File: day_3/part_1.rs
use std::fs;

fn main(){
  let input = fs::read_to_string("input.txt").unwrap();
  let lines = input.lines();
  let mut count: [i32; 12] = [0; 12];

  for line in lines {
    for (i, c) in line.chars().enumerate() {
      if c.to_digit(10).unwrap() > 0 {
        count[i] += 1;
      }else{
        count[i] -= 1;
      }
    }
  }

  let gamma = i32::from_str_radix(&count.map(|x| if x < 0 {'0'} else {'1'}).iter().collect::<String>(), 2).unwrap();
  let epsilon = i32::from_str_radix(&count.map(|x| if x < 0 {'1'} else {'0'}).iter().collect::<String>(), 2).unwrap();

  println!("gamma: {}\nepsilon: {}\ngamma * epsilon: {}", gamma, epsilon, gamma * epsilon)
}
